src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -939,29 +948,26 @@ export class StudyViewPageStore
             }
 
             // group present in page session which are not saved to user account
-            const missingGroupIds = Array.from(
-                this._selectedComparisonGroups.keys()
+            const missingGroupIds: string[] = Array.from(
+                this.createdGroups.keys()
             ).filter(groupId => groupIdSet[groupId] === undefined);
 
             if (missingGroupIds.length > 0) {
-                const promises = [];
-                for (const groupId of missingGroupIds) {
-                    promises.push(comparisonClient.getGroup(groupId));
-                }
+                const promises = missingGroupIds.map(groupId =>
+                    comparisonClient.getGroup(groupId)
```

Comment:
comment getGroup will make a new group if it doesn't exist?

Reply:
It doesn't make new group, if gets group by id if it is present.